docs/src/tutorial.md

```diff
@@ -35,18 +35,58 @@ norm(A \ v - op * v) / norm(v)
 ```
 In this example, the Cholesky factor is computed only once and can be used many times transparently.
 
+## mul!
+
+It is often useful to reuse the memory used by the operator. 
+For that reason, we can use `mul!` on operators as if we were using matrices
+using preallocated vectors:
+
+```@example ex2
+using LinearOperators, LinearAlgebra # hide
+m, n = 50, 30
+A = rand(m, n) + im * rand(m, n)
+op = LinearOperator(A)
+v = rand(n)
+res = zeros(eltype(A), m)
+res2 = copy(res)
+mul!(res2, op, v) # compile 3-args mul!
+al = @allocated mul!(res, op, v) # op * v, store result in res
+println("Allocation of LinearOperator mul! product = $al")
+v = rand(n)
+α, β = 2.0, 3.0
+mul!(res2, op, v, α, β) # compile 5-args mul!
+al = @allocated mul!(res, op, v, α, β) # α * op * v + β * res, store result in res
+println("Allocation of LinearOperator mul! product = $al")
```

Comment:
It was because the `res` value modifies the result of the 5-args `mul!`, so I wanted to keep `res` the same when calling the 5-args `mul!` for the first time. Should I remove it?